crates/pipeline-manager/src/db/operations/pipeline.rs

```diff
@@ -1320,6 +1321,15 @@ pub(crate) async fn set_deployment_resources_status(
         ResourcesStatus::Provisioning | ResourcesStatus::Provisioned => current.deployment_initial,
     };
 
+    let final_deployment_error = match final_deployment_error {
+        None => None,
+        Some(ref v) => Some(serialize_error_response(v)?),
+    };
+
+    let deployment_resources_status = new_deployment_resources_status.to_string(); // $7: deployment_resources_status,
```

Comment:
Can we keep the `final_` prefix here? The reason"final" is used is because it has determined whether it needs to be set to NULL or not.